tests/net/err.rs

```diff
@@ -0,0 +1,111 @@
+//! Test network errors
+//!
+//! The error implementation for the test networking code is slightly involved and has been factored
```

Comment:
Would this generally live elsewhere? If not, I don't think you need this explanation.